scintillometry/fold.py

```diff
@@ -0,0 +1,100 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import operator
+import warnings
+
+import numpy as np
+import astropy.units as u
+import astropy.time as time
+from .base import TaskBase
+
+
+__all__ = ['TimeFold',]
+
+#TODO add Fold base class for the future folding
+class TimeFold(TaskBase):
+    """ Fold pulse using a fixed time interval.
+
+    NOTE
+    ----
+    This method does not require the folding time to be the pulse period.
+    However, the time integration for each phase bin will not be uniform.
+
+    Parameters
+    ----------
+    ih : task or `baseband` stream reader
+        Input data stream, with time as the first axis.
+    n_phase : int
+        Number of bins in one result phase period.
+    phase : callable
+        The method to compute pulse phases at a given time.
+        Note, the input time is in the format of '~astropy.time.Time' object.
+        The output phase should be an array.
+    fold_time : float or `~astropy.units.Quantity`,
+        Time interval for folding into one pulse period(result sample).
+        Default unit is second.
```

Comment:
I think the default of "whole file" does make sense: gives you an average pulse profile, i.e.,
```
if fold_time is None:
    fold_time = ih.stop_time - ih.start_time
else:
    fold_time = u.Quantity(fold_time, u.s)
```